### PR TITLE
[Keyboard]meow48

### PR DESCRIPTION
**Pull request by tsubuan145**
## Types of Changes

- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by keyboard-magpie** [comment]
Can you actually use the PR template properly, please?

**Comment by tsubuan145** [comment]
Yes. Best regards.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please use Unix line endings (LF) for all files. Do not use the GitHub web interface to commit files, it will use Windows line endings (CRLF).
config.h, info.json, readme.md and both rules.mk files are all using CRLF.

**Comment by tsubuan145** [comment]
[meow48.zip](https://github.com/qmk/qmk_firmware/files/10041282/meow48.zip)
I fixed it. Please confirm.

**Comment by tsubuan145** [comment]
I fixed it again. Please confirm.

**Comment by tsubuan145** [comment]
I fixed it again. Please confirm.

**Review by fauxpark — CHANGES_REQUESTED** [review]
You still haven't fixed the line endings.

**Do not use the GitHub web UI or GitHub desktop to create keyboards.**

Set up the QMK environment, run `qmk new-keyboard`, and use `git` to push your changes.
https://docs.qmk.fm/#/newbs

**Comment by zvecr** [comment]
You might want to actually apply the requested fixes, before requesting review again.

**Comment by tsubuan145** [comment]
unknown_2022 10 01-18 30
unknown_2022 11 19-22 45
I fixed it. The key layout and schematic are like this.

**Comment by tsubuan145** [comment]
I will withdraw the pull request once because there are many corrections. I'm sorry.

**Comment by tsubuan145** [comment]
Various files have been modified.

**Comment by fauxpark** [comment]
These changes don't make sense. You've just added the keyboard again inside the original keyboard folder.

I would really recommend learning how to use git and GitHub first (as well as a decent code editor like VSCode or Sublime).

**Comment by tsubuan145** [comment]
I'm sorry. I uploarded the fille before the correction by mistake.

**Comment by tsubuan145** [comment]
I uploaded the modified file again.

**Review by fauxpark — CHANGES_REQUESTED** [review]
You still have not fixed the line endings; they need to be Unix style (LF), not Windows style (CRLF).

**Comment by tsubuan145** [comment]
Fixed a new line code.

**Comment by tsubuan145** [comment]
I'm sorry. The meow.h file has also been modified.

File: keyboards/meow48/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        KC_ESCAPE,  LT(3,KC_Q),    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, KC_BSPC,
+        KC_TAB,     KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,          KC_ENT,
+        KC_LSFT,    KC_Z,    LT(2,KC_X),    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT,
+        KC_CAPS,    KC_LCTL, KC_LGUI, KC_LALT,    KC_SPACE,               LT(1, KC_SPACE),   KC_DEL,  KC_LEFT, KC_UP,  KC_DOWN, KC_RGHT
+    ),
+    
+     [1] = LAYOUT(
+        KC_NO,  LSFT(1),    LSFT(2),    LSFT(3),    LSFT(4),    LSFT(5),    LSFT(6),    LSFT(7),    LSFT(8),    LSFT(9),  KC_LBRC, KC_CIRC, KC_BSPC,
+        KC_NO,     KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,          KC_ENT,
+        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_JYEN, KC_RO,         KC_RSFT,
```
Comment (fauxpark) [inline keyboards/meow48/keymaps/default/keymap.c:42]:
```suggestion
        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_INT3, KC_INT1,       KC_RSFT,
```

File: keyboards/meow48/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        KC_ESCAPE,  LT(3,KC_Q),    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, KC_BSPC,
+        KC_TAB,     KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,          KC_ENT,
+        KC_LSFT,    KC_Z,    LT(2,KC_X),    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT,
+        KC_CAPS,    KC_LCTL, KC_LGUI, KC_LALT,    KC_SPACE,               LT(1, KC_SPACE),   KC_DEL,  KC_LEFT, KC_UP,  KC_DOWN, KC_RGHT
+    ),
+    
+     [1] = LAYOUT(
+        KC_NO,  LSFT(1),    LSFT(2),    LSFT(3),    LSFT(4),    LSFT(5),    LSFT(6),    LSFT(7),    LSFT(8),    LSFT(9),  KC_LBRC, KC_CIRC, KC_BSPC,
+        KC_NO,     KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,          KC_ENT,
+        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_JYEN, KC_RO,         KC_RSFT,
```
Comment (fauxpark) [inline keyboards/meow48/keymaps/via/keymap.c:42]:
```suggestion
        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_INT3, KC_INT1,       KC_RSFT,
```

File: keyboards/meow48/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# meow48
+
+![meow48](imgur.com image replace me!)
```
Comment (fauxpark) [inline keyboards/meow48/readme.md:3]: Need an image here, or remove this line.

File: keyboards/meow48/rules.mk [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+MOUSEKEY_ENABLE = no
+EXTRAKEY_ENABLE = no
+CONSOLE_ENABLE = no
+COMMAND_ENABLE = no
+NKRO_ENABLE = no
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = yes
+AUDIO_ENABLE = no
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
```
Comment (fauxpark) [inline keyboards/meow48/rules.mk:15]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
OLED_ENABLE = yes
OLED_DRIVER = SSD1306
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by tsubuan145** [comment]
* { font-size: 13px; font-family: 'MS Pゴシック', sans-serif;}p, ul, ol, blockquote { margin: 0;}a { color: #0064c8; text-decoration: none;}a:hover { color: #0057af; text-decoration: underline;}a:active { color: #004c98;}
I fixed it. Please confirm.




Best regards.




----- Original Message -----




From: "Ryan" ***@***.***>

To: "qmk/qmk_firmware" ***@***.***>

Cc: "tsubuan_145" ***@***.***>; "State change" ***@***.***>

Date: 2022/11/21 月 01:33

Subject: Re: [qmk/qmk_firmware] [Keyboard]meow48 (PR #19106)







@fauxpark requested changes on this pull request.

In keyboards/meow48/keymaps/default/keymap.c:
> +     * │Caps │CTL│GUI│Alt│       │       │Del│LEFT│ UP │DOWN│RGHT│
+     * └─────┴───┴───┴───┴───────┴───────┴───┴────┴────┴────┴────┘
+     * 
+     */
+     
+     [0] = LAYOUT(
+        KC_ESCAPE,  LT(3,KC_Q),    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, KC_BSPC,
+        KC_TAB,     KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,          KC_ENT,
+        KC_LSFT,    KC_Z,    LT(2,KC_X),    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT,
+        KC_CAPS,    KC_LCTL, KC_LGUI, KC_LALT,    KC_SPACE,               LT(1, KC_SPACE),   KC_DEL,  KC_LEFT, KC_UP,  KC_DOWN, KC_RGHT
+    ),
+    
+     [1] = LAYOUT(
+        KC_NO,  LSFT(1),    LSFT(2),    LSFT(3),    LSFT(4),    LSFT(5),    LSFT(6),    LSFT(7),    LSFT(8),    LSFT(9),  KC_LBRC, KC_CIRC, KC_BSPC,
+        KC_NO,     KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,          KC_ENT,
+        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_JYEN, KC_RO,         KC_RSFT,
⬇️ Suggested change
-        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_JYEN, KC_RO,         KC_RSFT,
+        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_INT3, KC_INT1,       KC_RSFT,
In keyboards/meow48/keymaps/via/keymap.c:
> +     * │Caps │CTL│GUI│Alt│       │       │Del│LEFT│ UP │DOWN│RGHT│
+     * └─────┴───┴───┴───┴───────┴───────┴───┴────┴────┴────┴────┘
+     * 
+     */
+     
+     [0] = LAYOUT(
+        KC_ESCAPE,  LT(3,KC_Q),    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, KC_BSPC,
+        KC_TAB,     KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,          KC_ENT,
+        KC_LSFT,    KC_Z,    LT(2,KC_X),    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT,
+        KC_CAPS,    KC_LCTL, KC_LGUI, KC_LALT,    KC_SPACE,               LT(1, KC_SPACE),   KC_DEL,  KC_LEFT, KC_UP,  KC_DOWN, KC_RGHT
+    ),
+    
+     [1] = LAYOUT(
+        KC_NO,  LSFT(1),    LSFT(2),    LSFT(3),    LSFT(4),    LSFT(5),    LSFT(6),    LSFT(7),    LSFT(8),    LSFT(9),  KC_LBRC, KC_CIRC, KC_BSPC,
+        KC_NO,     KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,          KC_ENT,
+        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_JYEN, KC_RO,         KC_RSFT,
⬇️ Suggested change
-        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_JYEN, KC_RO,         KC_RSFT,
+        KC_LSFT,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_PLUS, KC_INT3, KC_INT1,       KC_RSFT,
In keyboards/meow48/readme.md:
> @@ -0,0 +1,27 @@
+# meow48
+
+meow48
Need an image here, or remove this line.

In keyboards/meow48/rules.mk:
> +MCU = atmega32u4
+
+BOOTLOADER = caterina
+
+BOOTMAGIC_ENABLE = no
+MOUSEKEY_ENABLE = no
+EXTRAKEY_ENABLE = no
+CONSOLE_ENABLE = no
+COMMAND_ENABLE = no
+NKRO_ENABLE = no
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = yes
+AUDIO_ENABLE = no
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
⬇️ Suggested change
-MCU = atmega32u4
-
-BOOTLOADER = caterina
-
-BOOTMAGIC_ENABLE = no
-MOUSEKEY_ENABLE = no
-EXTRAKEY_ENABLE = no
-CONSOLE_ENABLE = no
-COMMAND_ENABLE = no
-NKRO_ENABLE = no
-BACKLIGHT_ENABLE = no
-RGBLIGHT_ENABLE = yes
-AUDIO_ENABLE = no
-OLED_ENABLE = yes
-OLED_DRIVER = SSD1306
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no        # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306

File: keyboards/meow48/readme.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+*"Meow48" is a keyboard with a narrow pitch of 40%(48keys) exclusively for Kailh Choc.
```
Comment (zvecr) [inline keyboards/meow48/readme.md:1]:
```suggestion
# Meow48

"Meow48" is a keyboard with a narrow pitch of 40%(48keys) exclusively for Kailh Choc.
```

File: keyboards/meow48/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+     
+     [0] = LAYOUT(
+        KC_ESCAPE,  LT(3,KC_Q),    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, KC_BSPC,
+        KC_TAB,     KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,          KC_ENT,
+        KC_LSFT,    KC_Z,    LT(2,KC_X),    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT,
+        KC_CAPS,    KC_LCTL, KC_LGUI, KC_LALT,    KC_SPACE,               LT(1, KC_SPACE),   KC_DEL,  KC_LEFT, KC_UP,  KC_DOWN, KC_RGHT
+    ),
+    
+     [1] = LAYOUT(
+        KC_NO,  LSFT(1),    LSFT(2),    LSFT(3),    LSFT(4),    LSFT(5),    LSFT(6),    LSFT(7),    LSFT(8),    LSFT(9),  KC_LBRC, KC_CIRC, KC_BSPC,
```
Comment (zvecr) [inline keyboards/meow48/keymaps/default/keymap.c:40]: The use of `LSFT` here is invalid

File: keyboards/meow48/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+     
+     [0] = LAYOUT(
+        KC_ESCAPE,  LT(3,KC_Q),    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS, KC_BSPC,
+        KC_TAB,     KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN,          KC_ENT,
+        KC_LSFT,    KC_Z,    LT(2,KC_X),    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT,
+        KC_CAPS,    KC_LCTL, KC_LGUI, KC_LALT,    KC_SPACE,               LT(1, KC_SPACE),   KC_DEL,  KC_LEFT, KC_UP,  KC_DOWN, KC_RGHT
+    ),
+    
+     [1] = LAYOUT(
+        KC_NO,  LSFT(1),    LSFT(2),    LSFT(3),    LSFT(4),    LSFT(5),    LSFT(6),    LSFT(7),    LSFT(8),    LSFT(9),  KC_LBRC, KC_CIRC, KC_BSPC,
```
Comment (zvecr) [inline keyboards/meow48/keymaps/via/keymap.c:40]: The use of `LSFT` here is also invalid

File: keyboards/meow48/meow48.c [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#include "meow48.h"
+ 
+oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
+    return OLED_ROTATION_180;
+}
+
+#ifdef OLED_ENABLE
```
Comment (zvecr) [inline keyboards/meow48/meow48.c:23]:
```suggestion
#ifdef OLED_ENABLE

oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
    return OLED_ROTATION_180;
}
```

File: keyboards/meow48/meow48.h [hunk 9]
```diff
@@ -0,0 +1,44 @@
+/* Copyright 2022 tsubuan145
```
Comment (zvecr) [inline keyboards/meow48/meow48.h:1]: This file should be removed

File: keyboards/meow48/meow48.c [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#include "meow48.h"
```
Comment (zvecr) [inline keyboards/meow48/meow48.c:17]:
```suggestion
#include "quantum.h"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by tsubuan145** [comment]
I fixed it again. Please confirm.

Best regards.

**Comment by zvecr** [comment]
image

**Comment by tsubuan145** [comment]
I forgot to fix readme.md. I''m sorry.

**Comment by yswallow** [comment]
@tsubuan145 **YOU NEED TO REMOVE meow48.h**

**Comment by tsubuan145** [comment]
@yswallow OK

**Comment by yswallow** [comment]
:-1:

**Comment by zvecr** [comment]
`keyboards/meow48/meow48.h` still needs to be deleted.

**Comment by tsubuan145** [comment]
Removed meow48.h.

**Comment by tsubuan145** [comment]
The operation has been confirmed.

**Pull request closed without merging**